k9mail/src/main/java/com/fsck/k9/planck/PlanckProviderImplKotlin.kt

```diff
@@ -571,6 +579,63 @@ class PlanckProviderImplKotlin(
         }
     }
 
+    @WorkerThread
+    override suspend fun decryptMessage(
+        source: MimeMessage,
+        account: Account
+    ): Result<DecryptResult> = withContext(PlanckDispatcher) {
+        var srcMsg: Message? = null
+        var decReturn: decrypt_message_Return? = null
+        kotlin.runCatching {
+            //TODO review this; we are in another thread so we should get a new engine anyways??
+            srcMsg = PlanckMessageBuilder(source).createMessage(context)
+            val planckMsg = srcMsg!!
+            planckMsg.dir = Message.Direction.Incoming
+            planckMsg.recvBy = PlanckUtils.createIdentity(Address(account.email), context)
+
+            Timber.d("%s %s", TAG, "decryptMessage() before decrypt")
+            decReturn = engine.get().decrypt_message(srcMsg, Vector(), 0)
+            val decryptReturn = decReturn!!
+            Timber.d("%s %s", TAG, "decryptMessage() after decrypt")
+
+            when (decReturn!!.rating) {
```

Comment:
should we catch the possible exception and throw forward it further too?

Reply:
I see that later we are classifying this case as AppCannotDecryptException

Reply:
yes, it is just a wrapper, inside a runCatching anything goes more or less.